src/lichess.js

```diff
@@ -91,52 +95,93 @@ function getPlayerByName(name, isBackground){
+                    lichessRating.save().then(function() {
+                        console.log("Got updated rating for {name}: {rating}".format({
+                            name: name,
+                            rating: rating
+                        }));
+                        unlock.resolve();
+                    }).catch(function(error) {
+                        unlock.resolve();
+                    });
+                    return rating;
+                });
             });
-            _playerRatingPromises[name.toLowerCase()] = promise;
-        }
-
-        // Always return the latest one that we already have.
-        return Q.fcall(function() { return _playerRatings[name.toLowerCase()]; });
+        }).catch(function(error) {
+            console.error("Error getting player by name: " + error);
+        });
     }
-    function setPlayerRating(name, rating) {
-        _playerRatings[name.toLowerCase()] = rating;
+
+    //--------------------------------------------------------------------------
+    // Get the player rating. Always return the most recent rating from the
+    // database, unless they don't have it. Then go get it and return it.
+    //--------------------------------------------------------------------------
+    function getPlayerRating(name){
+        var name = name.toLowerCase();
+
+        // Get the writable lock for the database.
+        return db.lock().then(function(unlock) {
+            // Ensure we have a record.
+            return db.LichessRating.findOrCreate({
+                where: { lichessUserName: name }
+            }).then(function(lichessRating) {
+                unlock.resolve();
+                lichessRating = lichessRating[0];
+
+                var _30MinsAgo = moment.utc().subtract(30, 'minutes');
+                var lastCheckedAt = lichessRating.get('lastCheckedAt');
+                if (lastCheckedAt) {
+                    lastCheckedAt = moment.utc(lastCheckedAt);
+                }
+                var promise;
+                var isInQueue = _playerIsInQueue[name];
+                var rating = lichessRating.get('rating');
+
+                // Only update the rating if it's older than 30 minutes
+                // or if we don't have one a rating
+                // TODO: Replace this with _.isInteger
+                if (rating === null || typeof rating === 'undefined')  {
+                    // If we don't have a rating, use the foreground queue
+                    promise = _updateRating(name, false);
+                } else if (!isInQueue && (!lastCheckedAt || lastCheckedAt.isBefore(_30MinsAgo))) {
+                    // If the rating is just out of date, use the background queue
+                    promise = _updateRating(name, true);
+                }
+
+                if (rating !== null && typeof rating !== 'undefined')  {
+                    // Return the cached rating if we have it.
+                    return lichessRating.get('rating');
```

Comment:
I'm sorry, I don't understand your question here?  Promises are return until the end of the chain when a final "value" is returned. This is ending the chain by returning the final value.

Does that answer your question?